src/masonite/request.py

```diff
@@ -785,25 +785,33 @@ def reset_redirections(self):
         self.redirect_url = False
         self.redirect_route = False
 
-    def back(self, default=None):
+    def back(self, default=None, intended=False):
         """Return a URI for redirection depending on several use cases.
 
         Keyword Arguments:
             default {string} -- Default value if nothing can be found. (default: {None})
+            intended {boolean} -- Whether or not to redirect to a previously accessed route that required auth. (default: {False})
 
         Returns:
             self
         """
         self.with_input()
```

Comment:
Maybe I should move this to the else block of the `if intended` check so that the intended redirect doesn't get side effects of with_input